test/test_branch.ml

```diff
@@ -0,0 +1,112 @@
+let exit_success = 0
+let exit_failure = 1
+
+let operation = ref 0
+
+let logical_shift_right a b = incr operation ; a lsr b
+let logical_or a b = incr operation ; a lor b
+let shift_right a b = incr operation ; a asr b
+let logical_and a b = incr operation ; a land b
+let logical_not a = incr operation ; lnot a
+let minus a = incr operation ; (- a)
+let sub a b = incr operation ; a - b
+
+let[@inline always] minus_one_or_less n =
+  logical_shift_right n (sub Sys.int_size 1)
```

Comment:
Sys.int_size is a compile-time constant, so in all the cases I've seen the OCaml compiler will compute (63-1 = 62) and turn this into `logical_shift_right n 62`. The `Sys.int_size` is there for 32bit compatibility.